garak/generators/langchain_serve.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import json
+import requests
+
+from garak import _config
+from garak.generators.base import Generator
+
+class LangChainServeLLMGenerator(Generator):
+    """Class supporting LangChain Serve LLM interfaces via HTTP POST requests.
+
+    This class facilitates communication with LangChain Serve's LLMs through a web API, making it possible
+    to utilize external LLMs not directly integrated into the LangChain library. It requires setting up
+    an API endpoint using langchain serve.
+
+    Utilizes the HTTP POST method to send prompts to the specified LLM and retrieves the generated text
+    response. It is necessary to ensure that the API endpoint is correctly set up and accessible.
+
+    Inherits from Garak's base Generator class, extending its capabilities to support web-based LLM services.
+
+    Set endpoint name with --model_name. Example: --model_name=http://127.0.0.1:8000/rag-chroma-private and it's "invoke" endpoint will be used.
```

Comment:
model_name is used to specify `LangChainServeLLMGenerator`, can we avoid this being overloaded and set the URI in e.g. environment var? This will also be helpful to folks dealing with URI escaping on the CLI